docs/advanced_topics.rst

```diff
@@ -55,11 +55,11 @@ DALI provides a couple of ways to run a pipeline:
 - `schedule_run`, `share_outputs` and `release_outputs` with fine grain control of the output buffers' lifetime (corresponds to :meth:`nvidia.dali.types.PipelineAPIType.SCHEDULED` API type)
 - built-in iterators for MXNet, PyTorch, and TensorFlow (corresponds to :meth:`nvidia.dali.types.PipelineAPIType.ITERATOR` API type)
 
-The first API - `run` method launches the DALI pipeline, executing prefetch iterations if necessary, waits until the first batch is ready and returns the resulting buffers. Buffers are marked as in-use till the call to `run`. In many cases, it is wasteful as data is usually copied out to the native framework tensors after which they can be returned to DALI for reuse
+The first API - :meth:`nvidia.dali.pipeline.Pipeline.run` method launches the DALI pipeline, executing prefetch iterations if necessary, waits until the first batch is ready and returns the resulting buffers. Buffers are marked as in-use till the call to `nvidia.dali.pipeline.Pipeline.run`. In many cases, it is wasteful as data is usually copied out to the native framework tensors after which they can be returned to DALI for reuse
 
-The second API, consisting of `schedule_run`, `share_outputs` and `release_outputs` allows the user to explicitly manage the lifetime of the output buffers. The `schedule_run` method instructs DALI to prepare the next batch of data, doing prefetching if necessary. If the execution mode is set to asynchronous, this call returns immediately, without waiting for the results, so another task can be executed in parallel. The data batch can be requested from DALI by calling `share_outputs`, which returns the result buffer. If it is not ready yet, DALI will wait for it. The data is ready as soon as the `share_outputs` method returns. When DALI buffers are no longer needed, because data was copied or already consumed, `release_outputs` should be called to return DALI buffers for reuse in subsequent iterations.
+The second API, consisting of :meth:`nvidia.dali.pipeline.Pipeline.schedule_run`, :meth:`nvidia.dali.pipeline.Pipeline.share_outputs` and :meth:`nvidia.dali.pipeline.Pipeline.release_outputs` allows the user to explicitly manage the lifetime of the output buffers. The :meth:`nvidia.dali.pipeline.Pipeline.schedule_run` method instructs DALI to prepare the next batch of data, doing prefetching if necessary. If the execution mode is set to asynchronous, this call returns immediately, without waiting for the results, so another task can be executed in parallel. The data batch can be requested from DALI by calling `share_outputs`, which returns the result buffer. If it is not ready yet, DALI will wait for it. The data is ready as soon as the :meth:`nvidia.dali.pipeline.Pipeline.share_outputs` method returns. When DALI buffers are no longer needed, because data was copied or already consumed, :meth:`nvidia.dali.pipeline.Pipeline.release_outputs` should be called to return DALI buffers for reuse in subsequent iterations.
```

Comment:
```suggestion
The second API, consisting of :meth:`nvidia.dali.pipeline.Pipeline.schedule_run`, :meth:`nvidia.dali.pipeline.Pipeline.share_outputs` and :meth:`nvidia.dali.pipeline.Pipeline.release_outputs` allows the user to explicitly manage the lifetime of the output buffers. The :meth:`nvidia.dali.pipeline.Pipeline.schedule_run` method instructs DALI to prepare the next batch of data, prefetching if necessary. If the execution mode is set to asynchronous, this call returns immediately, without waiting for the results, so another task can be executed in parallel. The data batch can be requested from DALI by calling `share_outputs`, which returns the result buffer. If it is not ready yet, DALI will wait for it. The data is ready as soon as the :meth:`nvidia.dali.pipeline.Pipeline.share_outputs` method returns. When DALI buffers are no longer needed, because data was copied or already consumed, :meth:`nvidia.dali.pipeline.Pipeline.release_outputs` should be called to return DALI buffers to be reused in subsequent iterations.
```

Reply:
Ok, the diff is not very nice.
Changes:
1. `doing prefetching` -> `prefetching`
2. `for reuse` -> `to be reused`

Reply:
Done